src/main/java/org/opentripplanner/routing/edgetype/StreetEdge.java

```diff
@@ -555,21 +578,17 @@ public double getPredictedSpeed(long timeMillis) {
         return this.getMaxStreetTraverseSpeed();
     }
 
-    /**
-     * Calculate the speed appropriately given the RoutingRequest and traverseMode and the current wall clock time.
-     * Note: this is not strictly symmetrical, because in a forward search we get the speed based on the
-     * time we enter this edge, whereas in a reverse search we get the speed based on the time we exit
-     * the edge.
-     */
     public double calculateSpeed(RoutingRequest options, TraverseMode traverseMode, VehicleDescription currentVehicle, long timeMillis) {
         double maxVehicleSpeed = options.getSpeed(traverseMode);
-        if (currentVehicle != null) {
+        if(currentVehicle != null) {
             maxVehicleSpeed = currentVehicle.getMaxSpeedInMetersPerSecond(this);
         }
-        if (this.getTimes() != null && traverseMode == TraverseMode.CAR) {
-            return min(min(maxVehicleSpeed, getMaxStreetTraverseSpeed()), this.getPredictedSpeed(timeMillis));
+        if (this.getTimes()!= null && traverseMode== TraverseMode.CAR) {
+            return min(min(maxVehicleSpeed, getMaxStreetTraverseSpeed()), this.getVooomSpeed(timeMillis));
         }
-        return min(maxVehicleSpeed, getMaxStreetTraverseSpeed());
+        if (TemporarySpeedLimit>0)
```

Comment:
Tutaj nie można tak po prostu zwrócić `TemporarySpeedLimit`. Ten limit trzeba uwzględnić tylko, jeśli jedziemy autem `traverseMode == TraverseMode.CAR` i w dodatku nadal powinniśmy brać minimum z prędkości pojazdu, maksymalnej prędkości na danej ulicy i nowego limitu.

Reply:
to juz jest zapewnione przed wywolaniem  
public double calculateSpeed(RoutingRequest options, TraverseMode traverseMode, VehicleDescription currentVehicle, long timeMillis) {
        double maxVehicleSpeed = options.getSpeed(traverseMode);
        if(currentVehicle != null) {
            maxVehicleSpeed = currentVehicle.getMaxSpeedInMetersPerSecond(this);
        }
        if (this.getTimes()!= null && traverseMode== TraverseMode.CAR) {
            return min(min(maxVehicleSpeed, getMaxStreetTraverseSpeed()), this.getVooomSpeed(timeMillis));
        }